### PR TITLE
Ada language support

### PR DESCRIPTION
**Pull request by Mitchtdk**
Hi,
I have attempted to add support for Ada to the Compiler Explorer using the GNAT compiler. I have been able to get syntax highlighting working in the editor, all example programs compile and produce an assembly file that can be viewed. Currently the default result of compilation is just an assembly file, no binary is produced and there is no execution available. I have attempted to get this working locally, but GNAT is being a bit of a pain and I'm not an expert on using the GNAT compiler. It should however at least give someone who is a starting point.

I have based the ada.amazon.properties file on the fortran.amazon.properties file. I believe that GNAT should be included in GCC much like GFORTRAN if the build includes Ada as a language, if this is not the case let me know what I need to change.

I have run `make lint` and `make check` and everything appears to be passing for the Ada support.

Please let me know if there is anything else I may have missed.

### CODE REVIEW

**Comment by RabsRincon** [comment]
Hello @Mitchtdk! Thank you for taking the time to implement Ada to the CE supported languages, it looks really good, and the code is super nicely commented :)

We need to check that we do indeed have Ada available, or change some things around to get it. We'll get back to you as soon as possible with more info.

Thank you again for your contribution :)

File: examples/ada/Max_Array.adb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    type Integer_Array is array(Natural range <>) of Integer;
+    procedure Max_Array(x,y : in out Integer_Array) is
+    begin
+        for i in x'range loop
+            x(i) := (if (y(i) > x(i)) then y(i) else x(i));
+        end loop;
+    end Max_Array;
+begin
+    null;
+end Max_Array;
```
Comment (RabsRincon) [inline examples/ada/Max_Array.adb]: Newline missing at eof

File: examples/ada/Sum.adb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        sum : Natural := 0;
+    begin
+        for i in input'range loop
+            sum := sum + input(i);
+        end loop;
+        return sum;
+    end Sum;
+begin
+    null;
+end Sum;
```
Comment (RabsRincon) [inline examples/ada/Sum.adb]: Newline missing at eof

File: examples/ada/default.adb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+-- see : https://gcc.gnu.org/onlinedocs/gcc-8.2.0/gnat_rm/Pragma-Source_005fFile_005fName.html#Pragma-Source_005fFile_005fName
+pragma Source_File_Name (Square, Body_File_Name => "example.adb");
+
+-- Type your code here, or load an example.
+function Square(num : Integer) return Integer is 
+begin
+    return num**2;
+end Square;
+
+-- Ada 2012 also provides Expressiion Functions
```
Comment (RabsRincon) [inline examples/ada/default.adb]: Duplicated i in `Expresiion`, right?

File: examples/ada/default.adb [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+-- Type your code here, or load an example.
+function Square(num : Integer) return Integer is 
+begin
+    return num**2;
+end Square;
+
+-- Ada 2012 also provides Expressiion Functions 
+-- (http://www.ada-auth.org/standards/12rm/html/RM-6-8.html)
+-- as a short hand for functions whose body consists of a 
+-- single return statement. Hhowever they cannot be used as a
```
Comment (RabsRincon) [inline examples/ada/default.adb]: Typo. `However`

File: examples/ada/default.adb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+-- complication unit.
```
Comment (RabsRincon) [inline examples/ada/default.adb]: Typo `complication` instead of `compilation`

File: examples/ada/Max_Array.adb (hunk 1 above)
Comment (Mitchtdk) [reply examples/ada/Max_Array.adb]: Thanks for the review I have fixed up the newline and typos in the example projects.

**Comment by RabsRincon** [comment]
Hi @Mitchtdk, sorry for the lack of updates!

We currently do not build/have access to GCC's gnat binaries, but it should not be too hard to change the necessary bits over at the sister repo. 
If you want, feel free to open a PR there (mattgodbolt/compiler-explorer-image), otherwise, we will end up adding it ourselves, but it may take a while until we actually do add it (You might have noticed we're all kinda busy around here! :) )

**Comment by mattgodbolt** [comment]
I'm working on getting all the image changes in and run (at least for GCC trunk to start with), then we can merge. Thanks for your patience!

**Comment by mattgodbolt** [comment]
Will merge in now actually...and work on getting everything updated for the site! Again: thanks so much!!

**Comment by Mitchtdk** [comment]
@mattgodbolt great to hear, happy to have helped out.

**Pull request merged**